lib/src/HttpServer.cc

```diff
@@ -373,16 +384,29 @@ void HttpServer::onRequests(
             if (adviceFlag)
                 continue;
         }
+
+        // Optimization: Avoids dynamic allocation when copying the callback in
+        // handlers (ex: copying callback into lambda captures in DB calls)
+        auto paramPack = std::make_shared<CallBackParamPack>();
```

Comment:
I'm not sure if this is the best idea. But it is faster (likely because less data copying down the line). Yet it is very ugly. 

I've tried to make a wrapper class that wraps and allocates std::function in a shared_ptr. But that somehow turns out slower that the parameter pack. Source code: https://gist.github.com/marty1885/62b49260be6c1582d8cd632c986f57a8

Reply:
Also, benchmarks show putting `this` into the parameter pack doesn't seem to help even though that should avoid an allocation (small function optimization).